tests/integration/relationships/belongs-to-test.js

```diff
@@ -1725,3 +1732,75 @@ test("belongsTo relationship with links doesn't trigger extra change notificatio
 
   assert.equal(count, 0);
 });
+
+testRecordData(
+  "belongsTo relationship doesn't trigger when model data doesn't support implicit relationship",
+  function(assert) {
+    class TestModelData extends ModelData {
+      constructor(modelName, id, clientId, storeWrapper, store) {
+        super(modelName, id, clientId, storeWrapper, store);
+        delete this.__implicitRelationships;
+        delete this.__relationships;
+      }
+
+      _destroyRelationships() {}
+
+      _allRelatedModelDatas() {}
+
+      _cleanupOrphanedModelDatas() {}
+
+      _directlyRelatedModelDatas() {
+        return [];
+      }
+
+      destroy() {
+        this.isDestroyed = true;
+        this.storeWrapper.disconnectRecord(this.modelName, this.id, this.clientId);
+      }
+
+      get _implicitRelationships() {
+        return undefined;
+      }
+      get _relationships() {
+        return undefined;
+      }
+    }
+
+    Chapter.reopen({
+      book1: DS.belongsTo({ async: false }),
+    });
+
+    const createModelDataFor = env.store.createModelDataFor;
+    env.store.createModelDataFor = function(modelName, id, clientId, storeWrapper) {
+      if (modelName === 'book1') {
+        return new TestModelData(modelName, id, clientId, storeWrapper, this);
+      }
+      return createModelDataFor.call(this, modelName, id, clientId, storeWrapper);
+    };
+
+    run(() => {
+      env.store.push({
+        data: {
+          type: 'chapter',
+          id: '1',
+          relationships: {
+            book1: {
+              data: { type: 'book1', id: '1' },
+            },
+          },
+        },
+        included: [{ type: 'book1', id: '1' }],
+      });
+    });
+
+    let chapter = env.store.peekRecord('chapter', '1');
+    let book1 = env.store.peekRecord('book1', '1');
+
+    assert.ok(chapter.get('book1'));
+
+    // No inverse setup created for book1
+    // as Model-Data of book1 doesn't support this
+    // functionality.
+    assert.notOk(book1.get('chapter'));
```

Comment:
Since `chapter` is an implicit relationship, this would not work regardless. We should instead perform some actions that would create error if we have not guarded the feature correctly as well as assert that `_implicitRelationships` on the book1 modelData is indeed undefined.